src/systems/ec/implementations/vanilla/packages/ec-component-table/ec-component-table.js

```diff
@@ -0,0 +1,153 @@
+if (!Element.prototype.matches)
+  Element.prototype.matches = Element.prototype.msMatchesSelector;
+if (!Element.prototype.closest)
+  Element.prototype.closest = function poly(selector) {
+    let el = this;
+    while (el) {
+      if (el.matches(selector)) {
+        return el;
+      }
+      el = el.parentElement;
+    }
+    return null;
+  };
+
+/**
+ * @param {HTMLElement} element DOM element for component instantiation and scope
+ * @param {Object} options
+ * @param {String} options.sortSelector Selector for toggling element
+ * @param {Boolean} options.attachClickListener Whether or not to bind click events on toggle
+ */
+export class Table {
+  /**
+   * @static
+   * Shorthand for instance creation and initialisation.
+   *
+   * @param {HTMLElement} root DOM element for component instantiation and scope
+   *
+   * @return {Table} An instance of table.
+   */
+  static autoInit(root, { TABLE: defaultOptions = {} } = {}) {
+    const table = new Table(root, defaultOptions);
+    table.init();
+    // eslint-disable-next-line no-param-reassign
+    root.ECLTable = table;
+    return table;
+  }
+
+  constructor(element, { sortSelector = '[data-ecl-tablesort-toggle]' } = {}) {
+    // Check element
+    if (!element || element.nodeType !== Node.ELEMENT_NODE) {
+      throw new TypeError(
+        'DOM element should be given to initialize this widget.'
+      );
+    }
+
+    this.element = element;
+
+    // Options
+    this.sortSelector = sortSelector;
+
+    // Private variables
+    this.sortHeadings = null;
+
+    // Bind `this` for use in callbacks
+    this.handleClickOnSort = this.handleClickOnSort.bind(this);
+  }
+
+  /**
+   * @returns {HTMLElement}
+   */
+  static createSortIcon(customClass) {
+    const tempElement = document.createElement('span');
+    tempElement.innerHTML = iconSvgUiArrow; // avoiding the use of not-so-stable createElementNs
+    const svg = tempElement.children[0];
+    svg.removeAttribute('height');
+    svg.removeAttribute('width');
+    svg.setAttribute('focusable', false);
+    svg.setAttribute('aria-hidden', true);
+    // The following element is <path> which does not support classList API as others.
+    svg.setAttribute('class', `ecl-icon ecl-icon--2xs ${customClass}`);
+    return svg;
+  }
+
+  /**
+   * Initialise component.
+   */
+  init() {
+    this.sortHeadings = queryAll(this.sortSelector, this.element);
+
+    // Bind click event on toggles
+    if (this.sortHeadings) {
+      this.sortHeadings.forEach(tr => {
+        const sort = document.createElement('span');
+        sort.classList.add('ecl-table__arrow');
+        /* eslint-disable-next-line unicorn/prefer-node-append */
+        sort.appendChild(Table.createSortIcon('ecl-table__icon-up'));
+        /* eslint-disable-next-line unicorn/prefer-node-append */
+        sort.appendChild(Table.createSortIcon('ecl-table__icon-down'));
+        /* eslint-disable-next-line unicorn/prefer-node-append */
+        tr.appendChild(sort);
```

Comment:
wanted to use `prepend` here, but it's not IE compatible, so indeed of having a new Polyfill.. 
I use `appendChid` as the element is absolute positioning.

Reply:
Yes, `appendChild` is better for IE compatibility in my opinion as well.
I've actually removed this eslint plugin `unicorn` here https://github.com/ec-europa/europa-component-library/pull/1591/commits/cc104fdc5e545de3e2cb91e51964098d197ef8ae because its suggestions are too frequently irrelevant to the ECL code base patterns.

Reply:
When you have multiple occurrences of a rule which is irrelevant for a given file, you can disable the rule as following from the top of the file

```
/* eslint-disable unicorn/prefer-node-append */
```